elastic/kibana
# [Saved Objects] Add a root level managed property to all saved object documents

## Summary
Part of https://github.com/elastic/kibana/issues/140364


ATM, fleet uses a custom tag to identify saved objects installed during package installation. The current implementation has a few disadvantages, such as needing different tag ids per space, making it complex to determine if a saved object is really managed. 

As mentioned in https://github.com/elastic/kibana/issues/140364#issuecomment-1380636562, a better solution would be if there were some sort of property on a document that indicated if it was managed or not, which can then be read from the UI and the behavior toggled accordingly.

The property would default to false and allows the fleet plugin to set it to `true` on importing objects during package installation.

This PR:
1. adds an optional `managed` root-level property to all saved object documents
2. backfills/migrates `managed` on existing objects using a core transform
3. manually migrates sample data (sample data's already on 8.8.0).

I'll add the option to set `managed` to the import typescript API in a follow-up PR. The option will only be available to plugins and won't have a specific API.

PR deployed on Cloud [here](https://tinaheiligers-pr-154515-saved-objects-add-managed.kbndev.co/)

### How to test this:
- run the PR locally or using the [cloud deployment](https://tinaheiligers-pr-154515-saved-objects-add-managed.kbndev.co/)
- Install a sample dataset
- Navigate to Saved Objects Management and inspect any saved object (except a dataview, the inspect action renders in the Data View App). A simple SO such as [Global Settings [8.8.0]](https://tinaheiligers-pr-154515-saved-objects-add-managed.kbndev.co/app/management/kibana/settings) is easier.
- Ensure that the saved object has a `managed` property that is `false` (the default).
- You can repeat the inspect action for the other saved objects too and ensure they all have the default for `managed` set.
- Changing `managed` to `true` will be implemented in a follow-up PR where we limit being able to set this to Fleet alone, on package install.

## Checklist

### Note: 
Kibana needs to run through a migration to 8.8.0 to add `managed`. If already on 8.8.0 (dev and, potentially, the Observability clusters that periodically pull from `main`) the transform won't run. If it turns out to be an issue, we either have to wait for 8.9 to backfill existing SO’s (assuming we'll allow that), work around it by diverging the core version from the kibana version, or come up with another plan.
